src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -933,14 +921,22 @@ internal static class PatternConversionExtensions
 
     internal abstract class DecisionTree
     {
-        public DecisionTree Parent;
         public readonly BoundExpression Expression;
         public readonly TypeSymbol Type;
         public bool MatchIsComplete;
 
         public enum DecisionKind { ByType, ByValue, Guarded }
         public abstract DecisionKind Kind { get; }
 
+#if DEBUG
+        internal string Dump()
+        {
+            var builder = new StringBuilder();
+            DumpInternal(builder, 0);
+            return builder.ToString();
+        }
+        internal abstract void DumpInternal(StringBuilder builder, int indent);
```

Comment:
Yes, that would be much simpler. I'll do that.

---
In reply to: [69160615](https://github.com/dotnet/roslyn/pull/12250#discussion_r69160615) [](ancestors = 69160615)